### ISSUE TITLE
Compilation error: failed to select a version for the requirement `funty = "~1.2"`

### ISSUE THREAD

**Issue by vicsn** [microsoft/Nova#118]
Full compilation error:
```
error: failed to select a version for the requirement `funty = "~1.2"`
candidate versions found which didn't match: 2.0.0, 1.1.0, 1.0.1, ...
location searched: crates.io index
required by package `bitvec v0.22.3`
    ... which satisfies dependency `bitvec = "^0.22.3"` of package `ec-gpu-gen v0.3.0`
    ... which satisfies dependency `ec-gpu-gen = "^0.3.0"` of package `bellperson v0.22.0`
    ... which satisfies dependency `bellperson = "^0.22"` of package `bellperson-nonnative v0.3.1`
    ... which satisfies dependency `bellperson-nonnative = "^0.3.1"` of package `nova-snark v0.8.1 (/home/t490/programs/Nova)`
```

For an amusing history which led to this issue, I think this is relevant: https://github.com/bitvecto-rs/funty/issues/3

TL;DR, seems as if they killed funty version 1.2 5 days ago: `https://github.com/bitvecto-rs/funty/issues/3#issuecomment-1316243059`

I think using `bellperson = "0.24.0"` (and updating bellperson-nonnative to also use this bellperson) will solve it.

**Comment by srinathsetty** [comment]
Thanks, @vicsn! We have released a new version of the package that fixes the issue.
